### PR TITLE
chore: convert issue templates to issue forms

### LINKED ISSUE #78: [Feature] Convert current issue templates into issue forms
Github introduced **Issue Forms** a moment ago (still in beta though).
It is a better alternative to  static **Issue Templates** (in Markdown), as they
allow to better describe and above all, validate the information that must/should be provided when creating an new bug report or feature request.

Please convert the current templates under `.github/ISSUE_TEMPLATE/*` into this new format (YAML).

This might help: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms

### PR DESCRIPTION
**Pull request by aneeshrelan**
Closes #78

### CODE REVIEW

**Comment by aneeshrelan** [comment]
@tinesoft I tried converting the exact template to forms, not sure if you'd like some proper validations. Feel free to suggest

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Bug Report
+description: Create a report to help us improve
+title: "[Bug] "
+labels: [':bug: Bug']
+body:
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:5]: Please add the following description, before the actual inputs:
```
body:
  - type: markdown
    attributes:
      value: |
        Thanks for taking the time to fill out this bug report!
```

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+- type: input
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:6]: Separate this input into two:

* Plugin Name (id: pluginName) 
	* Use the *type*: dropdown
	* Use as possibles *options*: @nxrocks/spring-boot, @nxrocks/flutter, @nxrocks/quarkus
* Plugin Version (id: pluginVersion)
	* Add *placeholder* with value: `for e.g. v1.0.0`

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+name: Feature request
+description: Suggest an idea for this project
+title: "[Feature] "
+labels: [":new: enhancement"]
+body:
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:5]: Please add the following description, before the actual inputs:

```
body:
  - type: markdown
    attributes:
      value: |
        Thanks for taking the time to suggest this feature request!
```

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    label: Is your feature request related to a problem? Please describe
+    description: A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
+  validations:
+    required: false
+- type: textarea
+  attributes:
+    label: Describe the idea you'd like
+    description: A clear and concise description of what you want to happen
+  validations:
+    required: false
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:17]: change it to `require: true`

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    label: Plugin Name and Version
+  validations:
+    required: true
+- type: input
+  attributes:
+    label: Nx Version
+    description: "`nx --version`"
+  validations:
+    required: true
+- type: input
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:17]: Remove this entry, as `Nx Version` superseeds it

**Review by tinesoft — CHANGES_REQUESTED** [review]
Thanks for the PR!

In addition, 
* Please add a `id` field on all the *attributes*
* Remove the old `bug_report.md` and `feature_request.md` files

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+labels: [':bug: Bug']
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to fill out this bug report!
+  - type: dropdown
+    id: pluginName
+    attributes:
+      label: Plugin Name and Version
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:13]: remove "and Version" from label please

**Comment by aneeshrelan** [comment]
Thanks for taking a look @tinesoft. Have incorporated all requested changes, please check. Thanks! :)

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      options:
+        - "@nxrocks/spring-boot"
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:15]: Please add missing `"nx-"` prefix after each `@nxrocks/`  (my bad, I misspelled in in my first comment), as such:

```
- "@nxrocks/nx-spring-boot"
- "@nxrocks/nx-flutter"
- "@nxrocks/nx-quarkus"
```

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    validations:
+      required: true
+  - type: input
+    id: pluginVersion
+    attributes:
+      label: Plugin Version
+      placeholder: for e.g. v1.0.0
+    validations:
+      required: true
+  - type: input
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:27]: add `id: nxVersion`

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    attributes:
+      label: Nx Version
+      description: "`nx --version`"
+    validations:
+      required: true
+  - type: textarea
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:34]: add `id: expectedBehavior`

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    attributes:
+      label: Expected Behaviour
+      placeholder: ...
+    validations:
+      required: false
+  - type: textarea
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:41]: add `id: actualBehavior`

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+    attributes:
+      label: Actual Behaviour
+      placeholder: ...
+    validations:
+      required: false
+  - type: textarea
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:48]: add `id: reproductionSteps`

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+      description: A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
+    validations:
+      required: false
+  - type: textarea
+    attributes:
+      label: Describe the idea you'd like
+      description: A clear and concise description of what you want to happen
+    validations:
+      required: true
+  - type: textarea
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:24]: add `id: alternativeSolutions`

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    attributes:
+      label: Describe alternatives you've considered
+      description: A clear and concise description of any alternative solutions or features you've considered
+    validations:
+      required: false
+  - type: textarea
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:31]: add `id: additionalContext`

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+    attributes:
+      value: |
+        Thanks for taking the time to suggest this feature request!
+  - type: textarea
+    attributes:
+      label: Is your feature request related to a problem? Please describe
+      description: A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
+    validations:
+      required: false
+  - type: textarea
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:17]: add `id: ideaDescription`

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml [hunk 15]
```diff
@@ -0,0 +1,31 @@
+name: Feature request
+description: Suggest an idea for this project
+title: "[Feature] "
+labels: [":new: enhancement"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to suggest this feature request!
+  - type: textarea
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:10]: add `id: problemDescription`

**Review by tinesoft — CHANGES_REQUESTED** [review]
Looking good! Just a few things to adjust (see comments.

Also, please change the commit message from "feat: ..." to "chore: ...", as such:

`chore: convert issue templates to issue forms`

(

**Comment by aneeshrelan** [comment]
Updated. Please check. Also had the `id` added in another commit, slightly renamed a few as per your feedback

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 6 above)
Comment (tinesoft) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml:13]: You missed this!

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 7 above)
Comment (tinesoft) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml:15]: You missed this!

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 6 above)
Comment (aneeshrelan) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml]: Oops! My bad, missed pushing the commit and logged of 🙈

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+      description: "`nx --version`"
+    validations:
+      required: true
+  - type: textarea
+    id: expectedBehaviour
+    attributes:
+      label: Expected Behaviour
+      placeholder: ...
+    validations:
+      required: false
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:40]: please change this to `required: true`

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+  - type: textarea
+    id: actualBehaviour
+    attributes:
+      label: Actual Behaviour
+      placeholder: ...
+    validations:
+      required: false
```
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:47]: please change this to `required: true`

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 16 above)
Comment (tinesoft) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:40]: please change this to `required: true`

**Review by tinesoft — CHANGES_REQUESTED** [review]
OK, 2 more minor issues and we should be good!

**Comment by aneeshrelan** [comment]
Resolved @tinesoft

**Comment by tinesoft** [comment]
Thanks for your PR Aneesh! 🙏🏾 I merged it

**Pull request merged**